universal-login-sdk/lib/api/sdk.ts

```diff
@@ -134,13 +133,7 @@ class UniversalLoginSDK {
   }
 
   async getTokensDetails() {
-    const tokenDetails: TokenDetails[] = [];
-    for (const token of this.sdkConfig.observedTokens) {
-      const name = await this.tokenDetailsService.getName(token.address);
-      const symbol = await this.tokenDetailsService.getSymbol(token.address);
-      tokenDetails.push({...token, name, symbol});
-    }
-    return tokenDetails;
+    return Promise.all(this.sdkConfig.observedTokens.map(({address}) => this.tokenDetailsService.getTokenDetails(address)));
```

Comment:
`Promise.all` is a bad idea. It triggers concurrent requests that can flood the node/server. We can easily be kicked out from infura. In extreme cases might lead to resources depletion (mem, processes).

👍  for `using getTokenDetails()` thou

Reply:
1) Should be extracted, i.e.: tokenDetailsService.getTokenDetails
2) But even more importantly look loke sth that is already done somewhere else. Also, it should be stored so, one don't need to retrieve it every time. Perhaps you look for the following:
```ts
TokensDetailsStore.fetchTokensDetails()
```